front/src/common/InfraSelector/InfraSelectorModalBodyStandard.tsx

```diff
@@ -46,13 +46,14 @@ export default function InfraSelectorModalBodyStandard({
   const { t } = useTranslation(['translation', 'infraManagement']);
   const dispatch = useDispatch();
   const { mode } = useOsrdContext();
-  const updateInfraID = useUpdateInfraID();
+  const { updateInfraID, updateInfraLockStatus } = useInfraActions();
   const infraID = useInfraID();
   const { closeModal } = useContext(ModalContext);
   const { deleteItinerary } = useOsrdConfActions();
 
-  function setInfraID(id: number) {
+  function setInfraID(id: number, locked: boolean) {
```

Comment:
Change the function so it takes the infra as an arg
```suggestion
  function setInfraID(infra: Infra) {
```

Reply:
fixed